src/compiler.jl

```diff
@@ -3781,48 +3825,8 @@ function julia_allocator(B, LLVMType, Count, AlignedSize, IsDefault, ZI)
         else
             needs_dynamic_size_workaround = !isa(Size, LLVM.ConstantInt) || convert(Int64, Size) != 1
         end
-
-        T_size_t = convert(LLVM.LLVMType, Int; ctx)
-        if !needs_dynamic_size_workaround
-            # TODO Call declare_allocobj/emit_allocobj
-            if VERSION < v"1.8.0"
-                alloc_obj = get_function!(mod, "julia.gc_alloc_obj",
-                LLVM.FunctionType(T_prjlvalue, 
-                    [T_pint8, T_size_t, T_prjlvalue]))
-            else
-                alloc_obj = get_function!(mod, "julia.gc_alloc_obj",
-                    LLVM.FunctionType(T_prjlvalue, 
-                        [T_ppjlvalue, T_size_t, T_prjlvalue]))
-            end
-        else
-            # This doesn't allow for optimizations
-            alloc_obj = get_function!(mod, "jl_gc_alloc_typed",
-                LLVM.FunctionType(T_prjlvalue,
-                    [T_pint8, T_size_t, T_prjlvalue]))
-        end
-
-        if VERSION < v"1.7.0"
-            ptls = reinsert_gcmarker!(func, B)
-            ptls = bitcast!(B, ptls, T_pint8)
-        else
-            pgcstack = reinsert_gcmarker!(func, B)
-            ct = inbounds_gep!(B, 
-                bitcast!(B, pgcstack, T_ppjlvalue),
-                [LLVM.ConstantInt(current_task_offset(); ctx)])
-            ptls_field = inbounds_gep!(B, 
-                ct, [LLVM.ConstantInt(current_ptls_offset(); ctx)])
-            ptls = load!(B, bitcast!(B, ptls_field, T_ppint8))
-        end
-
-        if !needs_dynamic_size_workaround
-            if VERSION < v"1.8.0"
-                obj = call!(B, alloc_obj, [ptls, Size, tag])
-            else
-                obj = call!(B, alloc_obj, [ct, Size, tag])
-            end
-        else
-            obj = call!(B, alloc_obj, [ptls, Size, tag])
-        end
+        
+        obj = emit_allocobj!(B, tag, Size, needs_dynamic_size_workaround)
```

Comment:
Can we sink the needs_dynamic_workaround check?

Reply:
The first workaround (eg if constant or not), yes. The second workaround (whether there are gep users of a non-constant offset), no.